From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Tue, 16 Aug 2022 09:32:30 +0200
Subject: [PATCH] checkpatch: Print an error if rwlock.h is included directly.

rwlock.h shouldn't be included directly in source code. PREEMPT_RT uses
a different implementation and this rwlock.h include breaks it.

Add an error message if linux/rwlock.h is included.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Link: https://lkml.kernel.org/r/20220816075118.173455-1-bigeasy@linutronix.de
---
 scripts/checkpatch.pl |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/scripts/checkpatch.pl
+++ b/scripts/checkpatch.pl
@@ -5749,6 +5749,12 @@ sub process {
 			}
 		}
 
+# rwlock.h must not be included directly. It will be included via spinlock.h if needed.
+		if ($rawline =~ /\#\s*include.*\<linux\/rwlock\.h/) {
+			ERROR("INCLUDES_BAD_HEADER",
+				"Please don't include rwlock.h directly, use spinlock.h if needed.\n" . $herecurr);
+		}
+
 # warn if <asm/foo.h> is #included and <linux/foo.h> is available and includes
 # itself <asm/foo.h> (uses RAW line)
 		if ($tree && $rawline =~ m{^.\s*\#\s*include\s*\<asm\/(.*)\.h\>}) {
